src/components/CurrentWalletMagic.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from 'react'
+import * as rb from 'react-bootstrap'
+import { Link } from 'react-router-dom'
+import { useSettings } from '../context/SettingsContext'
+import { useCurrentWallet, useCurrentWalletInfo, useSetCurrentWalletInfo } from '../context/WalletContext'
+import Balance from './Balance'
+import Sprite from './Sprite'
+import { walletDisplayName } from '../utils'
+
+const WalletHeader = ({ name, balance, unit, showBalance }) => {
+  return (
+    <div className="d-flex flex-column align-items-center">
+      <h6 className="text-secondary">{walletDisplayName(name)}</h6>
+      <h4>
+        <Balance value={balance} unit={unit} showBalance={showBalance || false} />
+      </h4>
+    </div>
+  )
+}
+
+const PrivacyLevels = ({ accounts }) => {
+  const sortedAccounts = accounts.sort((lhs, rhs) => lhs.account - rhs.account).reverse()
+
+  return (
+    <div className="d-flex justify-content-center">
+      <div className="d-flex flex-column align-items-start" style={{ gap: '1rem' }}>
+        {sortedAccounts.map(({ account, account_balance: balance, branches }) => (
+          <PrivacyLevel key={account} level={parseInt(account)} balance={balance} />
+        ))}
+      </div>
+    </div>
+  )
+}
+
+const PrivacyLevel = ({ level, balance }) => {
+  const settings = useSettings()
+
+  const filledShields = Array(level + 1)
+    .fill()
+    .map((_, index) => {
+      return <Sprite key={index} symbol="shield-filled" width="24" height="30" />
+    })
+  const outlienedShields = Array(5 - filledShields.length)
```

Comment:
I just noticed that we can't always assume a mix depth of 5, since this can be configured by the user:

> The value of mixdepth runs from 0..M-1, where M is the number of mixdepths chosen by the user; by default 5.

See the [JoinMarket docs](https://github.com/JoinMarket-Org/JoinMarket-Docs/blob/master/High-level-design.md#wallets).

It's probably fine for now, but we might want to consider this at some point in the future.

Reply:
That is a very good point! It was an easy fix to make this dynamic so I figured I'd do it right away. We still use hard coded accounts in the Send/Receive views. Once we update those (in #57) we can make it dynamic there as well.